CMakeLists.txt

```diff
@@ -159,4 +189,52 @@ target_include_directories(monitor PRIVATE
   ${CMAKE_BINARY_DIR}
 )
 
-OPENDDS_TARGET_SOURCES(monitor std_qos.idl)
+add_executable(managed_testapp
+  ${MANAGER_HEADER}
+  test/managed.cpp
+)
+
+if (MSVC)
+  target_compile_definitions(managed_testapp PRIVATE _CRT_SECURE_NO_WARNINGS)
+else()
+  target_compile_features(managed_testapp PRIVATE cxx_std_17)
+endif()
+
+OPENDDS_TARGET_SOURCES(managed_testapp test/test.idl OPENDDS_IDL_OPTIONS "-Gxtypes-complete" SUPPRESS_ANYS OFF)
+
+target_link_libraries(managed_testapp
+  manager
+)
+
+target_include_directories(managed_testapp PRIVATE
+  ${CMAKE_BINARY_DIR}
+  src
+)
+
+add_executable(unmanaged_testapp
+  test/unmanaged.cpp
+)
+
+
+if (MSVC)
+  target_compile_definitions(unmanaged_testapp PRIVATE _CRT_SECURE_NO_WARNINGS)
+else ()
+  target_compile_features(unmanaged_testapp PRIVATE cxx_std_17)
+endif()
+
+OPENDDS_TARGET_SOURCES(unmanaged_testapp test/test.idl OPENDDS_IDL_OPTIONS "-Gxtypes-complete" SUPPRESS_ANYS OFF)
+
+target_link_libraries(unmanaged_testapp
+  OpenDDS::Dcps
+  OpenDDS::Rtps
+  OpenDDS::Rtps_Udp
+  OpenDDS::Tcp
+  OpenDDS::Shmem
+  OpenDDS::Udp
+  OpenDDS::Multicast
+)
+
+target_include_directories(unmanaged_testapp PRIVATE
+  ${CMAKE_BINARY_DIR}
```

Comment:
What is this including the binary directory?

Reply:
Doesn't seem to need it.  The `monitor` target does though, in order to find headers generated by qt5's tools.